background/sync-manager.js

```diff
@@ -40,9 +41,14 @@ const syncMan = (() => {
       {runNow: true});
   });
 
-  chrome.alarms.onAlarm.addListener(info => {
-    if (info.name === 'syncNow') {
-      syncMan.syncNow();
+  chrome.alarms.onAlarm.addListener(async ({name}) => {
+    if (name === 'syncNow') {
+      await syncMan.syncNow();
+      /* We want to fire the alarm once per SYNC_INTERVAL at a random point within the range,
+       * so the new delay includes the leftover portion from the last range. */
+      const r = Math.random();
+      schedule(SYNC_DELAY + SYNC_INTERVAL * (1 - lastInterval + r));
```

Comment:
Suppose the collision happens when two syncs start in 1 minute ~= 30mins * 0.033. The probability of the collision will be `1^2 - (1-0.033)^2 = 0.064911`, which is 1 collision every ~15.4 sync (1 collision every 7.7 hours).

I think it is better to use a fixed interval and implement something like exponential backoff.

Reply:
Maybe it'd be better to implement it in db-in-cloud?

Reply:
What's the verdict on this one? Is it something we should resolve before pushing a beta, or can it wait?

Reply:
Well, since db-in-cloud is still not fixed I guess I'll implement the workaround as suggested above. Today, hopefully.

Reply:
OTOH the probability of a collision is infinitely smaller because `delayInMinutes` is a fractional value, not an integer...

Reply:
@eight04, could you take a look at the new code?
@narcolepticinsomniac, I didn't test it, sync is ew 😝